### PR TITLE
Assignment1:parser and tests

### PR DESCRIPTION
**Pull request by SukainaShkeir**

### CODE REVIEW

File: src/model/Book.model.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    private orderId: number;
+    private bookTitle: string;
+    private author: string;
+    private genre: string;
+    private format: string;
+    private language: string;
+    private publisher: string;
+    private specialEdition: string;
+    private packaging: string;
+    private price: number;
```
Comment (samish-dev) [inline src/model/Book.model.ts:14]: price and quantity are part of the order

you should not include them in the item

Comment (samish-dev) [reply src/model/Book.model.ts:14]: same applies for the other items

File: src/model/Cake.model.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    private layers: number;
+    private frostingType: string;
+    private frostingFlavor: string;
+    private decorationType: string;
+    private decorationColor: string;
+    private customMessage: string;
+    private shape: string;
+    private allergies: string;
+    private specialIngredients: string;
+    private packagingType: string;
```
Comment (samish-dev) [inline src/model/Cake.model.ts:18]: notice that there is no price and quantity here in this item

File: src/model/Toy.model.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+export class Toy {
+    private OrderID: number;
+    private Type: string;
+    private AgeGroup: string;
+    private Brand: string;
+    private Material: string;
+    private BatteryRequired: boolean;
+    private Educational: boolean;
+    private Price: number;
+    private Quantity: number;
```
Comment (samish-dev) [inline src/model/Toy.model.ts:13]: keep the properties Camel case not Pascal case, and remove the god damn price and quantity

Comment (samish-dev) [reply src/model/Toy.model.ts:13]: sorry nothing personal, it's just a common mistake between the students

File: src/model/builders/book.builder.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    private bookTitle!: string;
+    private author!: string;
+    private genre!: string;
+    private format!: string;
+    private language!: string;
+    private publisher!: string;
+    private specialEdition!: string;
+    private packaging!: string;
+    private price!: number;
+    private quantity!: number;
```
Comment (samish-dev) [inline src/model/builders/book.builder.ts:15]: ...

File: src/model/builders/cake.builder.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+            this.type,
+            this.flavor,
+            this.filling,
+            this.size,
+            this.layers,
+            this.frostingType,
+            this.frostingFlavor,
+            this.decorationType,
+            this.decorationColor,
+            this.customMessage,
```
Comment (samish-dev) [inline src/model/builders/cake.builder.ts:102]: do we really want custom message as a required property?
in the data we saw that they could be null

File: tests/csvParser.test.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
+// src/__tests__/parsers/csvParser.test.ts
+
+import { parseCsv } from '../src/parsers/csvParser'; // Adjust path if needed
+import * as path from 'path';
+import * as fs from 'fs'; // Import fs for file operations during tests
+
+describe('CSV Parser', () => {
+    // --- Define Paths ---
+    // Assumes your test execution context makes __dirname relative to the test file
+    const dataDir = path.resolve(__dirname, '../../data'); // Path to actual data
```
Comment (samish-dev) [inline tests/csvParser.test.ts:10]: the data folder could potentially contain hundreds of thousands of lines
this is exhaustive and time consuming, a better approach is to create a data file in the tests directory that consists of max 5 lines to cover the edge cases

**Review by samish-dev — APPROVED** [review]
